Emphasize context/motivation of PRs for contributors (#677)
CitrineInformatics/citrine-python: CONTRIBUTING.md

* The linter finds no violations of PEP8 style
* Every line of code is executed by a test (100% coverage)

PR descriptions should describe the motivation and context of the code changes in the PR,
both for the reviewer and also for future developers.

> If there's a JIRA ticket or Github issue, linking to it is an easy and often effective way to provide that context.

Reviewer: I would say:
```suggestion
If there's a JIRA ticket or Github issue, the PR should be linked to the ticket/issue to provide that context.
```

Author: Personally, I'd rather be as outcome-driven as possible.  If you feel strongly, I'm OK with it, though.
